src/main/java/seedu/address/logic/commands/room/AllocateRoomCommand.java

```diff
@@ -26,90 +23,85 @@
 
-        if (!roomToEdit.isSameRoom(editedRoom) && model.hasRoom(editedRoom)) {
+        if (!roomToAllocate.isSameRoom(roomWithAllocatedPatient) && model.hasRoom(roomWithAllocatedPatient)) {
             throw new CommandException(MESSAGE_DUPLICATE_ROOM);
         }
 
-        model.setSingleRoom(roomToEdit, editedRoom);
-        return new CommandResult(String.format(MESSAGE_EDIT_ROOM_SUCCESS, editedRoom));
+        model.setSingleRoom(roomToAllocate, roomWithAllocatedPatient);
+        return new CommandResult(String.format(MESSAGE_ALLOCATE_ROOM_SUCCESS, roomWithAllocatedPatient));
     }
 
     /**
-     * Creates and returns a {@code Room} with the details of {@code roomToEdit}
-     * edited with {@code editRoomDescriptor}.
+     * Creates and returns a {@code Room} with the details of {@code roomToAllocate}
+     * edited with {@code allocateRoomDescriptor}.
      *
      * @param model Current model
-     * @param roomToEdit Room that is to be edited.
-     * @param editRoomDescriptor Details to edit the room with.
+     * @param roomToAllocate Room that is to be edited.
+     * @param allocateRoomDescriptor Details to edit the room with.
      * @return Room that has been edited.
      */
-    private Room createEditedRoom(Model model, Room roomToEdit,
-                                  EditRoomDescriptor editRoomDescriptor) throws CommandException {
-        assert (roomToEdit != null);
-        assert (editRoomDescriptor != null);
-
-        int updatedRoomNumber = editRoomDescriptor.getRoomNumber().orElse(roomToEdit.getRoomNumber());
-        TaskList roomTaskList = roomToEdit.getTaskList();
-        boolean isClearRoom = editRoomDescriptor.getIsOccupied().isPresent();
-        boolean hasNewPatient = editRoomDescriptor.getPatientName().isPresent();
-        boolean isCurrentlyOccupied = roomToEdit.isOccupied();
+    private Room allocatePatientToRoom(Model model, Room roomToAllocate,
+                                       AllocateRoomDescriptor allocateRoomDescriptor) throws CommandException {
+        assert (roomToAllocate != null);
+        assert (allocateRoomDescriptor != null);
+
+        Integer roomNumber = roomToAllocate.getRoomNumber();
+        TaskList roomTaskList = roomToAllocate.getTaskList();
+        boolean isClearRoom = allocateRoomDescriptor.getIsOccupied().isPresent();
+        boolean hasNewPatient = allocateRoomDescriptor.getPatientName().isPresent();
+        boolean isCurrentlyOccupied = roomToAllocate.isOccupied();
         if (!isClearRoom && !hasNewPatient && isCurrentlyOccupied) {
             //case 1: change room number in a room with patient already
-            return new Room(updatedRoomNumber, roomToEdit.getPatient(), roomTaskList);
+            return new Room(roomNumber, roomToAllocate.getPatient(), roomTaskList);
         } else if (!hasNewPatient) {
             //case 2: change room number in an empty room
```

Comment:
Same for this, do we still need this case since previously it was meant for change room number?